observability/metric/meter_test.go

```
@@ -0,0 +1,216 @@
+		t.Setenv("OTEL_EXPORTER_OTLP_ENDPOINT", "localhost:4317")
+		t.Setenv("OTEL_EXPORTER_OTLP_INSECURE", "true")
+
+		ctx := context.Background()
+		res := resource.Default()
+
+		mp, err := newMeterProvider(ctx, res)
+
+		// In unit test environment without OTLP collector, this may fail
+		// but we test that the function handles errors appropriately
+		if err != nil {
+			assert.Error(t, err)
```

golangci-lint (2.5.0)

[error] require-error: for error assertions use require

(testifylint)